src/test/java/htsjdk/samtools/seekablestream/SeekableStreamFactoryTest.java

```diff
@@ -69,7 +97,17 @@ public void testPathWithEmbeddedSpace() throws IOException {
             final int BYTES_TO_READ = 10;
             Assert.assertEquals(seekableStream.read(new byte[BYTES_TO_READ], 0,BYTES_TO_READ), BYTES_TO_READ);
         }
-
     }
 
+    @Test
+    public void testGetSeekableStreamWorksOnJimfs() throws IOException {
+        try(final FileSystem fs = Jimfs.newFileSystem(Configuration.unix())) {
+            final Path file = fs.getPath("/file");
+            Files.writeString(file,"hello");
+            try(final SeekableStream stream = SeekableStreamFactory.getInstance().getStreamFor(file.toUri().toString())){
+                final byte[] bytes = stream.readAllBytes();
+                Assert.assertEquals(new String(bytes, StandardCharsets.UTF_8), "hello");
+            }
+        }
+    }
```

Comment:
Is there any easy way to get test coverage for the case where `isBeingHandledByLegacyUrlSupport()` detects that an NIO plugin exists for the legacy-supported URL? I'm guessing not, but thought I'd ask.

Reply:
not an easy way, we can add an additional test suite in gradle but it's going to be a hassle.

Reply:
forget it then

Reply:
I did a manual test where I added the http-nio library to the test dependencies and isBeingHandledByLegacySupport for http / https is changes as expected.